app/components/search-facet-daterange/component.js

```diff
@@ -76,16 +89,21 @@ export default Ember.Component.extend({
         }
     }),
 
-    buildQueryObject(start, end) {
-        let key = this.get('key');
-        return dateRangeFilter(key, start, end);
+    buildQueryObject(start, end, dateType) {
+        return dateRangeFilter(dateType, start, end);
     },
 
-    updateFilter(start, end) {
-        let key = this.get('key');
-        let value = start && end ? { start: moment(start).format(DATE_FORMAT), end: moment(end).format(DATE_FORMAT) } : { start: '', end: '' };
+    updateFilter(start, end, dateType) {
+        const options = this.get('filterOptions');
+        for (const option in options) {
```

Comment:
Since the value isn't reassigned within the loop, `const` is fine ([for...of docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of)). I will update the `for...in` to `for...of` though.